### ISSUE TITLE
Always get 'Could not find a face that downscales correctly within epsilon' after running run.py

### ISSUE THREAD

**Issue by HelixNGC7293** [adamian98/pulse#19]
Hello Alex!
Just got this weird issue.
I indeed used _align_face.py_ to generate the input pics from realpics folder, but when I was using _run.py_ to upscale those pics, it keep saying 'Could not find a face that downscales correctly within epsilon'...
Did I do something wrong? Or is there any specific requirement for original mosaic photos?

**Comment by udf** [comment]
Not sure if this is the recommended way to fix this, but I had the same problem with a 32x32 image. I upscaled it to 1024x1024 (bicubic, so the result is blurry), and it seemed to work for me.

**Comment by MoArtis** [comment]
I got the same error with all the images I tried. But only one of them worked using @udf 's method. Thanks anyway! :)

**Comment by Jakorbie** [comment]
Increasing the number of steps would solve the issue.

**Comment by lspgn** [comment]
It seems to have been introduced by this commit https://github.com/adamian98/pulse/commit/ece9211c6448d3bca6c1d8c106ecfbf2e8eaed0f
I ran a test on Face-Depixelizer with it reverted, it worked.

**Comment by rbracco** [comment]
The above change adds a new argument `eps` to run.py such that if the L2 loss goes below eps, the program returns. If at the end of `steps` steps, the L2 loss is not less than eps the run fails with that error. Thus another fix is to choose a value for `eps` that is less than your L2 loss at the end. 

![image](https://user-images.githubusercontent.com/47190785/85290183-95233d80-b466-11ea-9834-73dfb7b37635.png)

After 100 steps I had an L2 loss of 0.006 (6e-3) so I set my eps to 8e-3 with the following colab code, and it worked. The step after this is to lower epsilon and increase steps to get a better result. The best option would be reverting though as we generally just want to see the best result achieved. 

```
seed = 100 #@param {type:"integer"}
epsilon = 8e-3
noise_type = 'trainable'  #@param ['zero', 'fixed', 'trainable']
optimizer = 'adam'  #@param ['sgd', 'adam','sgdm', 'adamax']
learning_rate = 0.4 #@param {type:"slider", min:0, max:1, step:0.05}
learning_rate_schedule = 'linear1cycledrop'  #@param ['fixed', 'linear1cycle', 'linear1cycledrop']
steps = 750 #@param {type:"slider", min:100, max:1000, step:50}
clear_output()

seed = abs(seed)
print('Estimated Runtime: {}s.\n'.format(round(0.23*steps)+6))
!python run.py \
  -seed $seed \
  -noise_type $noise_type \
  -opt_name $optimizer \
  -learning_rate $learning_rate \
  -steps $steps \
  -lr_schedule $learning_rate_schedule \
  -eps $epsilon
```

**Comment by 0xinit** [comment]
Hi @udf , how did you upscaled your images? I have my input in 128x128, and I was looking to upscale it to 1024x1024. Can you explain a bit?

**Comment by nanmehta** [comment]
hi...can anyone please help me know how the downsampling loss is implemented. Is it l1 or l2 loss being used to compare the generated low-resolution and the LR image.
please any1 let me know
